packages/blade/src/components/Tour/TourPopover.tsx

```diff
@@ -0,0 +1,178 @@
+  const isOppositeAxis = side === 'right' || side === 'bottom';
+
+  const [controllableIsOpen, controllableSetIsOpen] = useControllableState({
+    value: isOpen,
+    defaultValue: defaultIsOpen,
+    onChange: (isOpen) => onOpenChange?.({ isOpen }),
+  });
+
+  const { refs, floatingStyles, context, placement: computedPlacement } = useFloating({
+    open: controllableIsOpen,
+    onOpenChange: (isOpen) => controllableSetIsOpen(() => isOpen),
+    placement,
+    strategy: 'fixed',
+    middleware: [
+      shift({ crossAxis: false, padding: GAP }),
+      flip({ padding: GAP, fallbackAxisSideDirection: 'end' }),
+      offset(GAP + ARROW_HEIGHT),
+      arrow({
+        element: arrowRef,
+        padding: isHorizontal ? GAP + ARROW_HEIGHT : ARROW_WIDTH,
+      }),
+    ],
+    transform: true,
+    whileElementsMounted: autoUpdate,
+  });
+
+  const close = React.useCallback(() => {
+    controllableSetIsOpen(() => false);
+  }, [controllableSetIsOpen]);
+
+  // we need to animate from the offset of the computed placement
+  // because placement can change dynamically based on available space
+  const [computedSide] = getFloatingPlacementParts(computedPlacement);
+  const computedIsHorizontal = computedSide === 'left' || computedSide === 'right';
+  const animationOffset = isOppositeAxis ? -size[4] : size[4];
+
+  const { isMounted, styles } = useTransitionStyles(context, {
+    duration: {
+      open: transitionDelay,
+      close: theme.motion.duration.xquick,
+    },
+    initial: {
+      opacity: 0,
+      transform: `translate${computedIsHorizontal ? 'X' : 'Y'}(${animationOffset}px)`,
+    },
+  });
+
+  // remove click handler if popover is controlled
+  const isControlled = isOpen !== undefined;
+  const click = useClick(context, { enabled: !isControlled });
+  const dismiss = useDismiss(context);
+  const role = useRole(context);
+
+  const { getFloatingProps } = useInteractions([click, dismiss, role]);
+
+  const contextValue = React.useMemo(() => {
+    return {
+      close,
+      defaultInitialFocusRef,
+      titleId,
+    };
+  }, [close, titleId]);
+
+  // https://github.com/floating-ui/floating-ui/discussions/2352#discussioncomment-6044834
+  React.useLayoutEffect(() => {
+    window.setTimeout(() => {
+      if (!attachTo) return;
+      refs.setReference(attachTo.current);
+      refs.setPositionReference(attachTo.current);
+    });
+  }, [attachTo, refs, isOpen]);
+
+  return (
+    <PopoverContext.Provider value={contextValue}>
+      <FloatingPortal>
+        <FloatingFocusManager
+          disabled={isOpen === false || !isMounted}
+          initialFocus={defaultInitialFocusRef}
+          context={context}
+          modal={true}
+          guards={true}
+        >
+          {isMounted ? (
+            <BaseBox
+              ref={refs.setFloating}
+              style={{ ...floatingStyles, pointerEvents: isMounted ? 'auto' : 'none' }}
+              // TODO: Tokenize zIndex values
+              zIndex={zIndex}
```

Comment:
save this in the tokens file

Reply:
It's already coming from `popoverZIndex` from `../Popover/constants`